ios/Classes/MapboxMapController.swift

```diff
@@ -810,6 +810,25 @@ class MapboxMapController: NSObject, FlutterPlatformView, MGLMapViewDelegate, Ma
         return trackCameraPosition ? mapView.camera : nil
         
     }
+    /*
+    *  Scan layers from top to bottom and return the first matching feature
+    */
+    private func firstFeatureOnLayers(at: CGPoint) -> MGLFeature? {
+        guard let style = mapView.style else { return  nil}
+       
+        // get layers in order (featureLayerIdentifiers is unordered)
+        let clickableLayers = style.layers.filter{ layer in
+            return featureLayerIdentifiers.contains(layer.identifier)
+        }
+
+        for layer in clickableLayers.reversed() {
+            let features = mapView.visibleFeatures(at: at, styleLayerIdentifiers: [layer.identifier])
```

Comment:
I have approved. Let's hope `visibleFeatures` is fast enough to run on many layers in a loop, otherwise we would need to come up with something else

Reply:
one thing to note is that calling visibleFeatures with one layer is faster than calling it with n layers - it might not be a factor of n, but from testing it seems to be roughly n/2 

Given that setting and defining the base execution speed as a and the new one as b:
b = a * 2 / n * n = a * 2 

So the difference should stay consistent at factor of 2 regardless of the number of layers.

Reply:
It's implemented somewhere in C++ layer https://github.com/mapbox/mapbox-gl-native/blob/b8edc2399b9640498ccbbbb5b8f058c63d070933/src/mbgl/renderer/renderer.cpp#L44

Reply:
So i just found that the same happens with android. visibleFeatures or its java pendant queryRenderedFeatures return features out of order even if the inputs are ordered.
So querying them one by one is the only option. I will add the android code to this PR and setting this to draft while i do the changes

Reply:
It's probably because Set implementation used is HashSet and not LinkedHashSet that will keep order of insertion.

Reply:
I had converted it to a list (ordered by layers on the map) as a first test, but the Features still came out random.

It would look something like this `Feature from layer A`, `Feature from layer B,` `Feature from layer A`, ...